lib/workers/repository/reconfigure/index.ts

```diff
@@ -0,0 +1,157 @@
+import is from '@sindresorhus/is';
+import JSON5 from 'json5';
+import type { RenovateConfig } from '../../../config/types';
+import { validateConfig } from '../../../config/validation';
+import { logger } from '../../../logger';
+import { platform } from '../../../modules/platform';
+import { ensureComment } from '../../../modules/platform/comment';
+import { scm } from '../../../modules/platform/scm';
+import { getCache } from '../../../util/cache/repository';
+import { readLocalFile } from '../../../util/fs';
+import { getBranchCommit } from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { detectConfigFile } from '../init/merge';
+import {
+  deleteReconfigureBranchCache,
+  setReconfigureBranchCache,
+} from './reconfigure-cache';
+
+export function getReconfigureBranchName(prefix: string): string {
+  return `${prefix}reconfigure`;
+}
+export async function validateReconfigureBranch(
+  config: RenovateConfig
+): Promise<void> {
+  logger.debug('validateReconfigureBranch()');
+  const context = `renovate/config-validation`;
+
+  const branchName = getReconfigureBranchName(config.branchPrefix!);
+  const branchExists = await scm.branchExists(branchName);
+
+  // this is something the user initiates, so skip if no branch exists
+  if (!branchExists) {
+    logger.debug('No reconfigure branch found');
+    deleteReconfigureBranchCache(); // in order to remove cache when the branch has been deleted
+    return;
+  }
+
+  // look for config file
+  // 1. check reconfigure branch cache and use the configFileName if it exists
+  // 2. checkout reconfigure branch and look for the config file, don't assume default configFileName
+  const branchSha = getBranchCommit(branchName);
+  const cache = getCache();
+  let configFileName: string | null = null;
+  const reconfigureCache = cache.reconfigureBranchCache;
+
+  // only use valid cached information
+  if (reconfigureCache?.reconfigureBranchSha === branchSha) {
+    logger.debug('Cache is valid. Skipping validation check');
+    return;
+  }
+
+  try {
+    await scm.checkoutBranch(branchName);
+    configFileName = await detectConfigFile();
+  } catch (err) {
+    /*istanbul ignore next - should never happen*/
+    logger.error(
+      { err },
+      'Error while searching for config file in reconfigure branch'
+    );
+  }
+
+  if (!is.nonEmptyString(configFileName)) {
+    logger.warn('No config file found in reconfigure branch');
+    await platform.setBranchStatus({
+      branchName,
+      context,
+      description: 'Validation Failed - No config file found',
+      state: 'red',
+    });
+    setReconfigureBranchCache(branchSha!, false);
+    await scm.checkoutBranch(config.baseBranch!);
+    return;
+  }
+
+  let configFileRaw: string | null = null;
+  try {
+    configFileRaw = await readLocalFile(configFileName, 'utf8');
+  } catch (error) {
+    /*istanbul ignore next - should never happen*/
+    logger.error({ error }, 'Error while reading config file');
```

Comment:
```suggestion
  } catch (err) {
    /*istanbul ignore next - should never happen*/
    logger.error({ err }, 'Error while reading config file');
```

We've special logger handling for `err`. Also easily testable. 😉 
I would prefer to only skip coverage if it's really hard to test.